firestore.rules

```diff
@@ -45,6 +45,11 @@ service cloud.firestore {
     match /venues/{venue} {
       allow read: if true;
 
+      // Allow anyone to set the centrally project track.
+      // TODO Decide whether this belongs on the venue, or, somewhere else.
+      allow update: if (request.resource.data.diff(resource.data).affectedKeys()
+        .hasOnly(['projectedVideoTrackId']));
+
```

Comment:
The problem with just using roles is that this kind of logic will pop up in different places.

How will moving this data to a different place work?
Can you have a sub-collection or fields that aren't being read because you don't have the rule permission for them?
Does it require a function or does it work out of the box?

Reply:
This is just for a prototype. I have no intention of this going into production. Agree with all the concerns here.

I think a cloud function is the right move here.

Reply:
I'm unclear about the intent of the code itself. I'm guessing I will have to know some Twillio specifics like what a project video track is and how it relates to our use cases.

Maybe when ready (out of DRAFT) we can check this PR out with the preview link.

Reply:
https://7.sparkle.space/in/w/devplayground/s/experiment ;)